blocks/form/checkbox.js

```diff
@@ -0,0 +1,30 @@
+function update(checkboxgroup) {
+  const allUnchecked = checkboxgroup.every((checkbox) => !checkbox.checked);
+  checkboxgroup.forEach((checkbox) => {
+    if (allUnchecked) {
+      checkbox.setAttribute('required', '');
+    } else {
+      checkbox.removeAttribute('required');
+    }
+  });
+}
+
+export default async function decorate(form) {
+  const checkboxes = form.querySelectorAll('input[type=checkbox][required]');
+  const checkboxNameMap = [...checkboxes].reduce((map, checkbox) => {
+    if (map.has(checkbox.name)) {
+      map.get(checkbox.name).push(checkbox);
+    } else {
+      map.set(checkbox.name, [checkbox]);
+    }
+    return map;
```

Comment:
```
[...checkboxes].reduce((map, checkbox) => ({
   ...map
   [checkbox.name] : [checkbox, ...(map[checkbox.name] || [] )]   
}), {})
```

Reply:
using `getElementsByName` removing this code

Reply:
Will be using this implementation, since `getElementsByName` is available only on document and same name checkbox validation should be done by per form.

Reply:
done